assets/model_monitoring/components/src/shared_utilities/span_tree_utils.py

```diff
@@ -69,26 +78,156 @@ def attributes(self) -> str:
         """Get the span's attributes."""
         return self.get_node_attribute("attributes")  # type: ignore
 
+    @property
+    def events(self) -> str:
+        """Get the span's attributes."""
+        return self.get_node_attribute("events")  # type: ignore
+
     @property
     def input(self) -> str:
-        """Get the span's input from the attributes field as json string."""
+        """Get the span's input from the events payload field as json string."""
+        if self.events is None or len(self.events) == 0:
+            return self.input_from_attributes()
+        else:
+            events_array: list = json.loads(self.events)
+            if len(events_array) == 0:
+                return self.input_from_attributes()
+            input_event = list(filter(lambda event: event.get("name", None) == APP_TRACES_EVENT_LOG_INPUT_KEY,
+                                      events_array))
+            if input_event is not None and len(input_event) > 0:
+                if input_event[0].get("attributes", None) is None:
+                    return None
+                return input_event[0].get("attributes", None).get("payload", None)
```

Comment:
actually, it is a json string, I prefer to return the json.loads(json_str) instead of the json string, this will be friendly for the callers.

Reply:
The input and output is stored in mltable col as jsonstring format.

Reply:
yes, we have both use case to treat the inputs as jsonstring and the dict. So there is not significant benefit that make the input output as dict.